src/cloudConsole.ts

```diff
@@ -0,0 +1,217 @@
+import * as nls from 'vscode-nls';
+import * as path from 'path';
+import * as opn from 'opn';
+import * as cp from 'child_process';
+import * as semver from 'semver';
+import { TIMEOUT } from 'dns';
+import * as ws from 'ws';
+import * as fsExtra from 'fs-extra';
+
+const localize = nls.loadMessageBundle();
+
+export interface OS {
+	id: string;
+	shellName: string;
+	otherOS: OS;
+}
+
+export const OSes: Record<string, OS> = {
+	Linux: {
+		id: 'linux',
+		shellName: localize('azure-account.bash', "Bash"),
+		get otherOS() { return OSes.Windows; },
+	},
+	Windows: {
+		id: 'windows',
+		shellName: localize('azure-account.powershell', "PowerShell"),
+		get otherOS() { return OSes.Linux; },
+	}
+};
+
+export function openCloudConsole(api: AzureAccount, os: OS, files, outputChannel: OutputChannel) {
+	return (async function retry(): Promise<any> {
+
+		const isWindows = process.platform === 'win32';
+		if (isWindows) {
+			// See below
+			try {
+				const { stdout } = await exec('node.exe --version');
+				const version = stdout[0] === 'v' && stdout.substr(1).trim();
+				if (version && semver.valid(version) && !semver.gte(version, '6.0.0')) {
+					return requiresNode();
+				}
+			} catch (err) {
+				return requiresNode();
+			}
+		}
+
+		if (!(await api.waitForLogin())) {
+			return commands.executeCommand('azure-account.askForLogin');
+		}
+
+		const tokens = await Promise.all(api.sessions.map(session => acquireToken(session)));
+		const result = await findUserSettings(tokens);
+		if (!result) {
+			return requiresSetUp();
+		}
+
+		let consoleUri: string;
+		const armEndpoint = result.token.session.environment.resourceManagerEndpointUrl;
+		const inProgress = delayed(() => window.showInformationMessage(localize('azure-account.provisioningInProgress', "Provisioning {0} in Cloud Shell may take a few seconds.", os.shellName)), 2000);
+		try {
+			consoleUri = await provisionConsole(result.token.accessToken, armEndpoint, result.userSettings, os.id);
+			inProgress.cancel();
+		} catch (err) {
+			inProgress.cancel();
+			if (err && err.message === Errors.DeploymentOsTypeConflict) {
+				return deploymentConflict(retry, os, result.token.accessToken, armEndpoint);
+			}
+			throw err;
+		}
+
+		// TODO: How to update the access token when it expires?
+		let shellPath = path.join(__dirname, `../../bin/node.${isWindows ? 'bat' : 'sh'}`);
+		let modulePath = path.join(__dirname, 'cloudConsoleLauncher');
+		if (isWindows) {
+			modulePath = modulePath.replace(/\\/g, '\\\\');
+		}
+		const shellArgs = [
+			process.argv0,
+			'-e',
+			`require('${modulePath}').main()`,
+		];
+
+		if (isWindows) {
+			// Work around https://github.com/electron/electron/issues/4218 https://github.com/nodejs/node/issues/11656
+			shellPath = 'node.exe';
+			shellArgs.shift();
+		}
+
+		let response = await runInTerminal(result.token.accessToken, consoleUri);
+
+		// upload files to cloudshell
```

Comment:
this block is customized part

Reply:
could we keep customized part in a separate file?